pauthabielf64/pauthabielf64.rst

```diff
@@ -861,198 +740,313 @@ main can. The static linker must communicate the details of how to
 create the signed pointers by embedding the information in the ELF
 file. The format of the information is platform ABI as it is a
 contract between the static-linker and the C-runtime. One simple
-method of encoding the information is to create a dynamic relocation
-table as if dynamic linking as this contains all the necessary
-information. More compact encodings are possible.
+method of encoding the information is to emit a dynamic relocation
+section as if dynamic linking, with linker defined symbols denoting
+the base and limit of the section. The runtime can resolve the dynamic
+relocations to create the signed pointers. More compact encodings are
+possible.
 
 Run-time dynamic linking
 ========================
 
 On many platforms programs can load shared libraries at run-time via
-dlopen and access symbols in that library via ``dlsym`` or
+``dlopen`` and access symbols in that library via ``dlsym`` or
 ``dlvsym``. Some or all of these pointers may be signed. The signing
 schema for these functions is a platform decision that the compiled
-code and implementation of dlsym agree on. A simple implicit signing
-schema is for dlsym to sign code pointers but not data pointers. A
-more complex implemenation could add metadata to inform the dynamic
-linker how dlsym should sign the pointer. Finally variants of the
-dlsym, and dlvsym functions could be added with an extra parameter for
-the signing schema.
-
-TBD A mechanism to support different signing schemas for ``dlsym`` and
-``dlvsym``.
-
-Questions/Issues
+code and implementation of ``dlsym`` agree on.
 
-* Feedback on whether a signing schema is needed for runtime dynamic linking.
+The PAuth ABI uses a simple simple default signing schema. If the
+symbol found by ``dlsym`` has type ``STT_FUNC`` the address to be
+returned is signed with the ``IA`` key with a 0 modifier. Otherwise
+the address is not signed.
 
+An optional extension that communicates the signing schema for a
+symbol to the dynamic linker can be found in Appendix extension to
```

Comment:
Which appendix? Is that supposed to be a link?

Reply:
I've made it a link, have also fixed up a previous reference.